packages/angular/cli/models/schematic-command.ts

```diff
@@ -492,7 +492,7 @@ export abstract class SchematicCommand<
     }
 
     const pathOptions = o ? this.setPathOptions(o, workingDir) : {};
-    let input = Object.assign(pathOptions, args);
+    let input = { ...pathOptions, ...args };
```

Comment:
Without this change, I got the following TS error:
```
packages/angular/cli/models/schematic-command.ts:500:5 - error TS2322: Type '{ '--'?: string[] | undefined; } | { '--'?: string[] | undefined; }' is not assignable to type '{ [name: string]: string
; } & Arguments'.
  Type '{ '--'?: string[] | undefined; }' is not assignable to type '{ [name: string]: string; } & Arguments'.
    Type '{ '--'?: string[] | undefined; }' is not assignable to type '{ [name: string]: string; }'.
      Property ''--'' is incompatible with index signature.
        Type 'string[] | undefined' is not assignable to type 'string'.
          Type 'undefined' is not assignable to type 'string'.

500     input = {
```

As far as I can tell, that's because `Object.assign` makes an intersection type (`{ [name: string]: string; } & Arguments`, which doesn't let you have the undefined) while `{...something, ...somethingElse}` does a sort of union type instead.

Reply:
thanks for explaining